File: haskel/glue/src/Glue/Lib/Math/Power.hs
module Glue.Lib.Math.Power where

import Glue.Env qualified as E
import Glue.Eval (Eval)
import Glue.IR (Frame, IR (..), Native (..))
import Glue.Lib.Math.Power.Exp qualified as Exp
import Glue.Lib.Math.Power.Pow qualified as Pow
import Glue.Lib.Math.Power.Sqrt qualified as Sqrt

power :: Frame Eval
power =
    E.frameFromList
        [ ("exp", Native (Func Exp.exp))
        , ("pow", Native (Func Pow.pow))
        , ("sqrt", Native (Func Sqrt.sqrt))
        ]

File: dart/glue/lib/src/lib/math/power/power.dart
import 'package:glue/src/ir.dart';
import 'package:glue/src/module.dart';
import 'exp.dart';
import 'pow.dart';
import 'sqrt.dart';

/// Power module - exponential and power functions
/// Mirrors Haskell Glue.Lib.Math.Power exactly

/// The power module containing exponential and power operations
/// Mirrors Haskell Glue.Lib.Math.Power.power exactly
final ModuleInfo power = nativeModule('ffi.math.power', [
  // Exponential function
  ('exp', IrNativeFunc(exp)),

  // Power function
  ('pow', IrNativeFunc(pow)),

  // Square root function
  ('sqrt', IrNativeFunc(sqrt)),
]);
